### PR TITLE
Fix some tests for window

### PR DESCRIPTION
**Pull request by VivienDelmon**

### CODE REVIEW

File: Source/Common/gdcmDirectory.cxx [hunk 1]
```diff
@@ -73,8 +73,11 @@ unsigned int Directory::Explore(FilenameType const &name, bool recursive)
       }
     else
       {
-      Filenames.push_back(dirName+fileName);
-      nFiles++;
+      if (fileName[0] != '.') // discard "unix like" hidden files such as .git in submodules
```
Comment (malaterre) [inline Source/Common/gdcmDirectory.cxx]: I do not understand this change ? why should I skip hidden files ? Technically we would need to do the same in the `WIN32` section below. Also the change is not backward compatible, so I'd rather have a new option `SkipHiddenDotFile()`

File: Testing/Source/MediaStorageAndFileFormat/Cxx/TestPrinter1.cxx [hunk 2]
```diff
@@ -259,7 +259,13 @@ int TestPrint(const char *filename, bool verbose= false)
   std::ostringstream out;
   if( verbose )
     print.Print( std::cout );
+#if _MSC_VER <= 1800 // Visual Studio 2013
```
Comment (malaterre) [inline Testing/Source/MediaStorageAndFileFormat/Cxx/TestPrinter1.cxx]: I can cherry-pick this commit

File: Source/Common/gdcmDirectory.cxx (hunk 1 above)
Comment (VivienDelmon) [reply Source/Common/gdcmDirectory.cxx]: To be consistent with the Unix code that does that (line 120).
You already did that  for directories but did not applied it on files:
https://github.com/malaterre/GDCM/commit/6a627a55f72f65a5d7d98188dc46b274e500e7a1
The TestScanner1 is failing due to a .git file that is added in Testing/Data and which is not expected by the test

Comment (malaterre) [reply Source/Common/gdcmDirectory.cxx]: For reference line 120 points to commit 9e2791a9, 6a627a5 is about the hidden directory part.

**Pull request merged**